langchain/src/memory/zep_memory.ts

```diff
@@ -0,0 +1,252 @@
+import {
+  ZepClient,
+  SearchResult,
+  SearchPayload,
+  Memory,
+  Message,
+  MessageData,
+  NotFoundError,
+} from "@getzep/zep-js";
+
+import {
+  BaseChatMessageHistory,
+  BaseChatMessage,
+  AIChatMessage,
+  HumanChatMessage,
+} from "../schema/index.js";
+
+import { InputValues, MemoryVariables, getBufferString } from "./base.js";
+import { BaseChatMemory, BaseChatMemoryInput } from "./chat_memory.js";
+
+/**
+ * A ChatMemoryHistory implementation that uses Zep as backend Memory store for conversations.
+ * Recommended Usage::
+ *  Setup a Zep Chat History
+ *  const zepChatHistory = new ZepChatMessageHistory(sessionID, ZEP_API_URL);
+ *
+ *  Use a standard ConversationBufferMemory to encapsulate the Zep chat history
+ *  const memory = new ConversationBufferMemory("chat_history", zepChatHistory);
+ *
+ *  Zep provides long-term conversation storage for LLM apps. The server stores,
+ *  summarizes, embeds, indexes, and enriches conversational AI chat
+ *  histories, and exposes them via simple, low-latency APIs.
+ *
+ *  For server installation instructions and more, see: https://getzep.github.io/
+ *  This class is a thin wrapper around the zep-python package. Additional
+ *  Zep functionality is exposed via the zepSummary and zepMessages properties.
+ *
+ *  For more information on the zep-python package, see:
+ *  https://github.com/getzep/zep-js
+ *
+ */
+export class ZepChatMessageHistory extends BaseChatMessageHistory {
```

Comment:
I think this should live in `src/stores/zep.ts`? Could you theoretically use Zep with other message stores like Redis with the proper adapter?

Message stores are generally intended to be simple storage - can we move more of the processing into the memory class?

https://github.com/hwchase17/langchainjs/blob/main/.github/contributing/integrations/MESSAGE_STORES.md

Reply:
zep_memory is implemented as a memory class - since we summarize and process the chat histories (not just store them).

Reply:
Done